cmd/iam-store.go

```diff
@@ -2170,6 +2185,15 @@ func (store *IAMStoreSys) UpdateServiceAccount(ctx context.Context, accessKey st
 		cr.Comment = opts.comment
 	}
 
+	if opts.expiration != nil {
+		expirationInUTC := opts.expiration.UTC()
+		currentTime := time.Now().UTC()
+		if !expirationInUTC.IsZero() && !expirationInUTC.Equal(timeSentinel) && expirationInUTC.Before(currentTime) {
```

Comment:
We can just do:
```suggestion
		if opts.expiration.Before(time.Now()) {
```

As we might expect, all `time.Time` values have a location and so all this conversion is unnecessary. We also do not want to allow time.Zero or the timeSentinel as valid expiration values.

Reply:
We need the conversion because we use UTC values for credential expiry [checks](https://github.com/minio/minio/blob/master/internal/auth/credentials.go#L128)

```go
// IsExpired - returns whether Credential is expired or not.
func (cred Credentials) IsExpired() bool {
	if cred.Expiration.IsZero() || cred.Expiration.Equal(timeSentinel) {
		return false
	}

	return cred.Expiration.Before(time.Now().UTC())
}
```

Reply:
> We also do not want to allow time.Zero or the timeSentinel as valid expiration values.

I have added upper and lower bounds to expiration now, these checks are not required